src/main/java/org/jenkinsci/test/acceptance/po/WorkflowMultiBranchJob.java

```diff
@@ -0,0 +1,64 @@
+package org.jenkinsci.test.acceptance.po;
+
+import com.google.inject.Injector;
+import org.apache.commons.io.IOUtils;
+import org.jenkinsci.test.acceptance.plugins.workflow_multibranch.BranchSource;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A pipeline multi-branch job (requires installation of multi-branch-project-plugin).
+ *
+ */
+@Describable("org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject")
+public class WorkflowMultiBranchJob extends FolderItem {
+
+    public WorkflowMultiBranchJob(Injector injector, URL url, String name) {
+        super(injector, url, name);
+    }
+
+    public <T extends BranchSource> T addBranchSource(Class<T> type) {
+        ensureConfigPage();
+
+        control(by.path("/hetero-list-add[sources]")).selectDropdownMenu(type);
+
+        final By branchSourceConfiguration = by.xpath("//div[@name='sources']");
```

Comment:
If you are going to use name better to use `by.name` IMHO, even better if you can use `by.path`, if you can use `createPageArea` this code is no longer needed